pkg/nfd-master/nfd-master.go

```diff
@@ -277,13 +287,67 @@ func (m *nfdMaster) Run() error {
 			if m.nfdController != nil && m.args.EnableNodeFeatureApi {
 				m.nfdController.updateAllNodesChan <- struct{}{}
 			}
+			// Restart the node updater pool
+			m.stopNodeUpdaterPool()
+			go m.runNodeUpdaterPool()
+
 		case <-m.stop:
 			klog.Infof("shutting down nfd-master")
 			return nil
 		}
 	}
 }
 
+func (m *nfdMaster) processNodeLabelRequest() bool {
+	nodeName, quit := m.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer m.queue.Done(nodeName)
+
+	if err := m.nfdAPIUpdateOneNode(nodeName.(string)); err != nil {
+		if m.queue.NumRequeues(nodeName) < 5 {
+			klog.Infof("retrying labeling request for node %s", nodeName)
+			m.queue.AddRateLimited(nodeName)
+			return true
+		} else {
+			klog.Errorf("error labeling node %v: %v", nodeName, err)
+		}
+	}
+	m.queue.Forget(nodeName)
+	return true
+}
+
+func (m *nfdMaster) runNodeUpdater() {
+	for {
+		select {
+		case <-m.nodeUpdaterPoolStop:
+			return
+		default:
+			m.processNodeLabelRequest()
+		}
+	}
```

Comment:
Now that I'm looking at this I think it doesn't work at all: processNodeLabelRequest blocks and in practice the stop signal is probably never received (in particular as m.nodeUpdaterPoolStop is re-initialized before we get to the select again) -> growing number of leaking goroutines on every re-configuration event -> bad

I think the right thing to do is to shutdown the queue in runNodeUpdaterPool so that all blocking get operations get cancelled.
```suggestion
	for m.processNodeLabelRequest() {
	}
```